ui/app/src/Utils/FormValidation.elm

```diff
@@ -0,0 +1,62 @@
+module Utils.FormValidation
+    exposing
+        ( ValidationResult(Valid, NotValid)
+        , ValidatedString
+        , validateDate
+        , areValidResults
+            , validatedMatcherToMatcher
+        , ValidatedMatcher
+        )
+
+import Utils.Date exposing (timeFromString)
+import Silences.Types exposing (Silence)
+import Utils.Types exposing (Matcher)
+
+
+type ValidationResult
+    = Valid
+    | NotValid String
+
+
+type alias ValidatedString =
+    { value : String
+    , result : ValidationResult
+    }
```

Comment:
I have rewritten it, this time with the help of Elm `Result`.

Each value in the `SilenceForm` is now a `Result`. The `Result` for `startsAt` e.g. is: 
```
Result (String, String) (String, Time)
```
If  `startsAt` can be parsed it will be `Ok (String, Time)` where `String` is the string representation of `startsAt` and `Time` is the parsed time. This way whenever we want to access the parsed representation we don't have to parse it again.

If  `startsAt` can not be parsed it will be `Err (String, String)` where the first `String` is the string representation of `startsAt` and the second `String` is the error message explaining why it could not be parsed.

@w0rm @stuartnelson3 What do you think?
(@w0rm using `Result.map5` as suggested by you. Cool hint)